### PR TITLE
TEAM TEACHING BRANCH: First steps running the engine and the scene in the application to successfully render nothing

### PR DESCRIPTION
**Pull request by samcake**
- Insert a "main3DScene" and the "renderEngine" in the Application
 Init the engine with the default Tasks for now

- During Application::displaySide call the scene::processPendingChanges and then in sequence run the engine one frame

- one very first pass insert a Dummy render::Item/Payload and make sure it gets "rendered" correclty

With that we have a very first draft of the scene being rendered by the engine through the combination of Taks.
To be continued by checking that the update mechanism works
And then add true items to render

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: f0ab6d96f5a07794ae9c22edd9b6dbab5285844f available at: https://gist.github.com/6726b96a3592500fd01f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4963.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4963.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: ad054a640edc8d789b116c7ce9b4841a528be963 available at: https://gist.github.com/6726b96a3592500fd01f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4963.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4963.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0083ba6f857ab8d320ad40a1f18742ec66886a5b available at: https://gist.github.com/6726b96a3592500fd01f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4963.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4963.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: dd9d758da1e2da1d20d393235bf5071c06da2825 available at: https://gist.github.com/6726b96a3592500fd01f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4963.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4963.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: ad0b0cfade280d089c3cdbbd3da4f66e15858f5f available at: https://gist.github.com/6726b96a3592500fd01f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4963.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4963.dmg

**Comment by samcake** [comment]
SO this one should be good for review / merge

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -3157,6 +3161,25 @@ const ViewFrustum* Application::getDisplayViewFrustum() const {
...
 
+class MyFirstStuff {
+public:
+    typedef render::Payload<MyFirstStuff> Payload;
+    typedef std::shared_ptr<render::Item::PayloadInterface> PayloadPointer;
+    typedef Payload::DataPointer Pointer;
+        
+};
+
+// For Ubuntu, the compiler want's the Payload's functions to be specialized in the "render" namespace explicitely...
```
Comment (ZappoMan) [inline interface/src/Application.cpp]: I think you could also just use the render:: prefix on all your classes and templates... but I think your current approach is cleaner

**Comment by hifi-gustavo** [comment]
Build console output for commit: e5755f52e55817c8a615907a6d5066a4c074307b available at: https://gist.github.com/6726b96a3592500fd01f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4963.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4963.dmg

**Pull request merged**